src/main/java/com/redhat/developers/msa/aloha/AlohaVerticle.java

```diff
@@ -153,24 +169,34 @@ private void alohaChaining(RoutingContext context, Handler<List<String>> resultH
      * @return The feign pointing to the service URL and with Hystrix fallback.
      */
     private BonjourService getNextService(RoutingContext context) {
-        final String serviceName = "bonjour";
-        // This stores the Original/Parent ServerSpan from ZiPkin.
-        final ServerSpan serverSpan = (ServerSpan) context.data().get("zipkin.span");
-        final CloseableHttpClient httpclient =
-            HttpClients.custom()
-                .addInterceptorFirst(new BraveHttpRequestInterceptor(BRAVE.clientRequestInterceptor(), new DefaultSpanNameProvider()))
-                .addInterceptorFirst(new BraveHttpResponseInterceptor(BRAVE.clientResponseInterceptor()))
-                .build();
-        final String url = String.format("http://%s:8080/", serviceName);
+        String url = System.getenv("BONJOUR_SERVER_URL");
+        if (null == url || url.isEmpty()) {
+            url = String.format("http://%s:%s", System.getenv("BONJOUR_SERVICE_HOST"), System.getenv("BONJOUR_SERVICE_PORT"));
+        }
+
+        // as we are calling a service somewhere else, it's a good idea to instrument the request
+        // with the trace information, so that we can get a distributed tracing information!
+        // we stored the spanContext for this request on the context.data, so, we retrieve it first
+        SpanContext spanContext = (SpanContext) context.data().get("tracing.requestSpan");
+        Tracer tracer = TracerResolver.getTracer();
+
+        final CloseableHttpClient httpclient;
+        if (tracer != null) {
+            Span span = tracer.buildSpan("GET").asChildOf(spanContext).start();
+
+            // note that we add our two interceptors to the http client, so that we can add our trace state
+            // and mark the trace as finished once we get the answer
+            httpclient = HttpClients.custom()
```

Comment:
is `BonjourService getNextService(RoutingContext context)` called for each request? So it creates new http client for every call?